server/services/applicationService.ts

```diff
@@ -56,18 +56,29 @@ export default class ApplicationService {
     return page
   }
 
-  save(page: TasklistPage, request: Request) {
+  async save(page: TasklistPage, request: Request) {
     const errors = page.errors ? page.errors() : []
+
     if (errors.length) {
       throw new ValidationError(errors)
     } else {
-      request.session = this.fetchOrInitializeSessionData(request.session, request.params.task, request.params.id)
-
-      request.session.application[request.params.id][request.params.task][request.params.page] = page.body
-      request.session.previousPage = page.name
+      this.saveToSession(page, request)
+      await this.saveToApi(request)
     }
   }
 
+  private saveToSession(page: TasklistPage, request: Request) {
+    request.session = this.fetchOrInitializeSessionData(request.session, request.params.task, request.params.id)
+    request.session.application[request.params.id][request.params.task][request.params.page] = page.body
+    request.session.previousPage = page.name
+  }
+
+  private async saveToApi(request: Request) {
```

Comment:
I think we only want to save to the API at the end of a task, rather than update each time

Reply:
Can I ask why?
We'd have to build in some extra functionality to only save to the API after the end of a task whereas I can't see an issue saving after every response - but I could be missing something!

Reply:
Thanks for asking! I guess I was thinking it was to prevent any partially saved applications from being sent to the API, but actually, that's probably not the end of the world!

Reply:
I think we would be able to retrieve a partial application and repopulate the form. If it breaks anything we can add the 'only save to API at end of task' functionality later?

Reply:
👍🏻